internal/db/rules.go

```diff
@@ -1,175 +1,79 @@
 package db
 
 import (
-	"fmt"
-
-	"log"
-	"strconv"
-
-	"encoding/json"
-
+	"database/sql"
+	"github.com/pkg/errors"
 	// Blank import required for mysql driver
 	_ "github.com/go-sql-driver/mysql"
 	"github.com/jimmyjames85/bouncecm/internal/models"
 )
 
 // ListRules - Function to pull all rules from db
-func (c *Client) ListRules() (*models.RulesObject, error) {
+func (c *Client) GetAllRules() (*models.RulesObject, error) {
 	rules := []models.BounceRule{}
-
 	rows, err := c.Conn.Query("SELECT * FROM bounce_rule")
 
 	if err != nil {
-		log.Println(err)
-		fmt.Println("FAILEDHERE?")
-
-		return nil, err;
-
+		return nil, errors.Wrap(err, "GetAllRules Query")
 	}
 
+	defer rows.Close()
+
 	for rows.Next() {
 		br := models.BounceRule{}
-		var description *string
-		err := rows.Scan(&br.ID, &br.ResponseCode, &br.EnhancedCode, &br.Regex, &br.Priority, &description , &br.BounceAction)
-		
-		// case when description is null becuase it can be null according to droprules.sql
-		if description == nil {
-			br.Description = ""
-		} else {
-			br.Description = *description
-		}
 
+		var description sql.NullString
+		err := rows.Scan(&br.ID, &br.ResponseCode, &br.EnhancedCode, &br.Regex, &br.Priority, &description, &br.BounceAction)
 
+		if description.Valid {
+			br.Description = description.String
+		} else {
+			br.Description = ""
+		}
 		if err != nil {
-			log.Println(err)
-			fmt.Println(br)
-
-			return nil, err;
+			return nil, errors.Wrap(err, "GetAllRules Scanning")
 		}
-
 		rules = append(rules, br)
-		
-	}
-
-	rulesObject := models.RulesObject{Rules: rules, NumRules: len(rules)}
-
-	return &rulesObject, nil
-}
-
-func (c *Client) CreateRuleDB(rule *models.BounceRule) ([]byte, error) {
-	stmt, err := c.Conn.Prepare("INSERT INTO bounce_rule(response_code,enhanced_code,regex,priority,description,bounce_action) VALUES(?,?,?,?,?,?)")
-	
-	if err != nil {
-		log.Println(err)
-		return nil, err;
-	}
-
-
-	_, err = stmt.Exec(rule.ResponseCode, rule.EnhancedCode, rule.Regex, rule.Priority, rule.Description, rule.BounceAction)
-	
-	if err != nil {
-		log.Println(err)
-		return nil, err;
 	}
 
-	data, err := json.Marshal(rule)
-	
+	err = rows.Err()
 	if err != nil {
-		log.Println(err)
-		return nil, err;
+		return nil, errors.Wrap(err, "GetAllRules Row.Err")
 	}
-	return data, nil
-}
 
-func (c *Client) GetRuleDB(id string) (*models.BounceRule, error) {
-	var bounce_rule *models.BounceRule
-	rows, err := c.Conn.Query("SELECT * FROM bounce_rule WHERE id=" + id)
-	for rows.Next() {
-		var br models.BounceRule
-		err = rows.Scan(&br.ID, &br.ResponseCode, &br.EnhancedCode, &br.Regex, &br.Priority, &br.Description, &br.BounceAction)
-		if err != nil {
-			log.Println(err)
-			return nil, err;
-		}
-		bounce_rule = &br
-		return bounce_rule, nil
-	}
-	return nil, err
+	rulesObject := models.RulesObject{Rules: rules, NumRules: len(rules)}
```

Comment:
I'm still not sure why `RulesObject` struct has `NumRules`. Why not just simply return the slice `[]models.BounceRule{}`?

Reply:
this was part of the initial commit months ago. I will change it now